codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -178,22 +191,26 @@ class ServerProtocolTestGenerator(
         writeInline("let expected =")
         instantiator.render(this, inputShape, httpRequestTestCase.params)
         write(";")
-        with(httpRequestTestCase) {
-            host.orNull()?.also { host ->
-                val withScheme = "http://$host"
-                rustTemplate(
-                    """
-                    let mut http_request = http_request;
-                    let ep = #{SmithyHttp}::endpoint::Endpoint::mutable(#{Http}::Uri::from_static(${withScheme.dq()}));
-                    ep.set_endpoint(http_request.uri_mut(), parts.acquire().get());
-                    """,
-                    *codegenScope,
-                )
-            }
-            resolvedHost.orNull()?.also { host ->
-                rust("""assert_eq!(http_request.uri().host().expect("host should be set"), ${host.dq()});""")
+        httpRequestTestCase.body.orNull()?.also { body ->
+            rustTemplate(
+                """
+                ##[allow(unused_mut)] let mut http_request = http::Request::builder()
+                    .uri(${httpRequestTestCase.uri.dq()})
+                    .body(#{SmithyHttpServer}::Body::from(#{Bytes}::from_static(b${body.dq()}))).unwrap();
+                """,
+                *codegenScope
+            )
+            if (!httpRequestTestCase.bodyMediaType.isEmpty()) {
+                rust("""http_request.headers_mut().insert("Content-Type", http::header::HeaderValue::from_static(${httpRequestTestCase.bodyMediaType.get().dq()}));""")
             }
         }
+        if (!httpRequestTestCase.queryParams.isEmpty()) {
+            val queryParams = httpRequestTestCase.queryParams.joinToString(separator = "&")
+            rust("""*http_request.uri_mut() = "${httpRequestTestCase.uri}?$queryParams".parse().unwrap();""")
+        }
+        httpRequestTestCase.host.orNull()?.also {
+            rust("// TODO: support endpoint trait")
```

Comment:
```suggestion
            rust("""todo!("endpoint trait not supported yet")""")
```

I'm thinking that we should emit errors in the log or preferably fail the build to alert the user if they make use of any of these yet-to-be-implemented features.

Reply:
Let me see if this has any impact we are unaware off right now.